### PR TITLE
chore: fix unnameable-types lint on `blockchain-tree` and `rpc` crates

### PR DESCRIPTION
**Pull request by TropicalDog17**
## Description

Partially addresses #9401 

Fix unnameable-types lint on `blockchain-tree` and `rpc` crates

### CODE REVIEW

File: crates/blockchain-tree/src/lib.rs [hunk 1]
```diff
@@ -56,4 +56,6 @@ pub mod noop;
 
 mod state;
 
+pub use state::BlockchainId;
```
Comment (mattsse) [inline crates/blockchain-tree/src/lib.rs:59]: I'd like to keep this type private or pub crate

**Review by mattsse — CHANGES_REQUESTED** [review]
ty, 1 nit

File: crates/blockchain-tree/src/lib.rs (hunk 1 above)
Comment (TropicalDog17) [reply crates/blockchain-tree/src/lib.rs:59]: Thanks @mattsse I will fix,  I'm curious about the reasoning behind this.

Comment (mattsse) [reply crates/blockchain-tree/src/lib.rs:59]: this type is only used internally and should not be exposed

**Review by emhane — APPROVED** [review]

**Review by mattsse — APPROVED** [review]

**Pull request merged**